### PR TITLE
Get Meetings (events) started

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: application/models/meeting.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	Description nulls.String `json:"description" db:"description"`
+	MoreInfoURL nulls.String `json:"more_info_url" db:"more_info_url"`
+	StartDate   time.Time    `json:"start_date" db:"start_date"`
+	EndDate     time.Time    `json:"end_date" db:"end_date"`
+	CreatedAt   time.Time    `json:"created_at" db:"created_at"`
+	UpdatedAt   time.Time    `json:"updated_at" db:"updated_at"`
+	CreatedByID int          `json:"created_by_id" db:"created_by_id"`
+	ImageFileID nulls.Int    `json:"image_file_id" db:"image_file_id"`
+	LocationID  int          `json:"location_id" db:"location_id"`
+
```
Comment (Schparky) [inline application/models/meeting.go]: Good, I like the separation here.

File: application/models/meeting.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		&validators.StringIsPresent{Field: m.Name, Name: "Name"},
+		&validators.TimeIsPresent{Field: m.StartDate, Name: "StartDate"},
+		&validators.TimeIsPresent{Field: m.EndDate, Name: "EndDate"},
+		&validators.IntIsPresent{Field: m.CreatedByID, Name: "CreatedByID"},
+		&validators.IntIsPresent{Field: m.LocationID, Name: "LocationID"},
+	), nil
+}
+
+// ValidateCreate gets run every time you call "pop.ValidateAndCreate" method.
+// This method is not required and may be deleted.
```
Comment (Schparky) [inline application/models/meeting.go]:
```suggestion
```

File: application/models/meeting.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+func (m *Meeting) ValidateCreate(tx *pop.Connection) (*validate.Errors, error) {
+	return validate.NewErrors(), nil
+}
+
+// ValidateUpdate gets run every time you call "pop.ValidateAndUpdate" method.
+// This method is not required and may be deleted.
```
Comment (Schparky) [inline application/models/meeting.go]:
```suggestion
```

File: application/models/meeting.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+// FindByUUID finds a meeting by the UUID field and loads its CreatedBy field
+func (m *Meeting) FindByUUID(uuid string) error {
+	if uuid == "" {
+		return errors.New("error finding message: uuid must not be blank")
+	}
+
+	queryString := fmt.Sprintf("uuid = '%s'", uuid)
+
+	if err := DB.Eager("CreatedBy").Where(queryString).First(m); err != nil {
```
Comment (Schparky) [inline application/models/meeting.go]:
```suggestion
	if err := DB.Eager("CreatedBy").Where("uuid = ?", uuid).First(m); err != nil {
```

File: application/models/meeting.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+	return nil
+}
+
+// AttachImage assigns a previously-stored File to this Meeting as its image. Parameter `fileID` is the UUID
+// of the image to attach.
+func (m *Meeting) AttachImage(fileID string) (File, error) {
+	var f File
+	if err := f.FindByUUID(fileID); err != nil {
+		return f, nil // in case client can't recognize this error we'll fail silently
```
Comment (Schparky) [inline application/models/meeting.go]: I see this is how I did it with `Post#AttachPhoto` but I wonder if now that we've improved error handling and the front-end error handling is currently being improved, maybe this should fail with an error.

File: application/models/meeting.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+	"time"
+)
+
+// Meeting represents an event where people gather together from different locations
+type Meeting struct {
+	ID          int          `json:"id" db:"id"`
+	UUID        uuid.UUID    `json:"uuid" db:"uuid"`
+	Name        string       `json:"name" db:"name"`
+	Description nulls.String `json:"description" db:"description"`
+	MoreInfoURL nulls.String `json:"more_info_url" db:"more_info_url"`
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/meeting.go]: @fillup I just took this from the Trello card.  Do you have a preference?

File: application/migrations/20191216141441_create_meetings.up.fizz [hunk 7]
```diff
@@ -0,0 +1,23 @@
+create_table("meetings") {
+	t.Column("id", "integer", {primary: true})
+	t.Column("uuid", "uuid", {})
+	t.Column("name", "string", {})
+	t.Column("description", "character varying(4096)", {null: true})
+	t.Column("more_info_url", "string", {null: true})
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/migrations/20191216141441_create_meetings.up.fizz]: @fillup I just took this from the Trello card. Do you have a preference?

[the remark this answers is not in the document]
Comment (Baggerone) [reply application/migrations/20191216141441_create_meetings.up.fizz]: Phillip said he prefers it as I had it.

File: application/migrations/20191216141441_create_meetings.up.fizz [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+	t.Column("start_date", "date", {})
+	t.Column("end_date", "date", {})
+	t.ForeignKey("created_by_id", {"users": ["id"]}, {"on_delete": "cascade"})
+	t.ForeignKey("image_file_id", {"files": ["id"]}, {"on_delete": "set null"})
+	t.ForeignKey("location_id", {"locations": ["id"]}, {"on_delete": "cascade"})
+	t.Timestamps()
+}
+
+add_index("meetings", "uuid", {"unique": true})
+add_index("meetings", "location_id", {"unique": true})
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/migrations/20191216141441_create_meetings.up.fizz]: But would we want to re-use the location record for multiple meetings?

**Pull request merged**